app/src/main/kotlin/com/intive/picover/profile/view/ProfileScreen.kt

```diff
@@ -31,8 +38,43 @@ fun ProfileScreen(viewModel: ProfileViewModel) {
 		Button(onClick = viewModel::onLogoutClick) {
 			Text(stringResource(R.string.LogoutButton))
 		}
-		Button(onClick = viewModel::onDeleteAccountClick) {
+		Button(
+			onClick = {
+				isDeleteAccountDialogVisible = true
+			},
+		) {
 			Text(stringResource(R.string.DeleteAccountButton))
 		}
 	}
+	if (isDeleteAccountDialogVisible) {
+		DeleteAccountDialog(
+			onConfirm = {
+				viewModel.onDeleteAccountClick()
+				isDeleteAccountDialogVisible = false
+			},
+			onDismiss = { isDeleteAccountDialogVisible = false },
+		)
+	}
+}
+
+@Composable
+private fun DeleteAccountDialog(
```

Comment:
Maybe this dialog could be more generic, at this moment I don't know where we can reuse it or if we can be worried about it this moment, maybe when someone would like to add again alert dialog will change it to a more generic 😄

Reply:
Like you said – we can think about making it generic when there will be a need for that 😄